readthedocs/search/parsers.py

```diff
@@ -111,21 +111,16 @@ def _get_main_node(self, html):
           so they are children of the same parent node.
         - Return the body element itself if all checks above fail.
         """
-        body = html.body
-        main_node = body.css_first("[role=main]")
-        if main_node:
-            return main_node
```

Comment:
I know that we had role=main for old versions of sphinx, they may see some extra content indexed now.

Reply:
I wasn't able to fix the tests when using `AddonsConfig.DEFAULT_ROOT_SELECTOR`, so I reverted my changes from this file. I will leave this for a next iteration. Maybe you know why they are failing after this change and can fix them.